.github/workflows/build-xcframework-packages.yml

```diff
@@ -0,0 +1,25 @@
+name: "Build XCFramework Packages"
+
+on:
+  workflow_call: # Allows calling from other workflows
+  workflow_dispatch: # Allows manual triggering
+
+jobs:
+  build-xcfranework-packages:
+    runs-on: macos-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Dependencies
+        uses: ./.github/actions/setup-dependencies
+
+      - name: Build XCFramework Packages
+        run: make OPTIMIZE=yes PLATFORMS="all" CONFIGS="static" srcs
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: xcframework-packages
+          path: cpp/lib/XCFrameworks/*
```

Comment:
This will create on zip with all the frameworks. When we do a tag we'll want each xcframeworks to be in its own zip. 

https://developer.apple.com/documentation/xcode/distributing-binary-frameworks-as-swift-packages

Might be easier to separate that here for when we want to upload tag artifacts.